File: apps/entity-service/src/entities/enums.ts
export {
  EntityType,
  EntitySource,
  Classification,
} from '@sentinel/proto-gen';
import { Classification } from '@sentinel/proto-gen';

// These enums live in @sentinel/shared-models (ESM) but entity-service is CJS,
// so we define them locally to avoid CJS/ESM interop issues.

export enum TrackEnvironment {
  AIR = 'AIR',
  SEA_SURFACE = 'SEA_SURFACE',
  SUBSURFACE = 'SUBSURFACE',
  GROUND = 'GROUND',
  SPACE = 'SPACE',
  UNKNOWN = 'UNKNOWN',
}

export enum TrackProcessingState {
  LIVE = 'LIVE',
  PREDICTED = 'PREDICTED',
  DEAD_RECKONED = 'DEAD_RECKONED',
  HYPOTHESIZED = 'HYPOTHESIZED',
  HISTORICAL = 'HISTORICAL',
}

export enum NavigationalStatus {
  UNDER_WAY_USING_ENGINE = 'UNDER_WAY_USING_ENGINE',
  AT_ANCHOR = 'AT_ANCHOR',
  NOT_UNDER_COMMAND = 'NOT_UNDER_COMMAND',
  RESTRICTED_MANOEUVRABILITY = 'RESTRICTED_MANOEUVRABILITY',
  CONSTRAINED_BY_DRAUGHT = 'CONSTRAINED_BY_DRAUGHT',
  MOORED = 'MOORED',
  AGROUND = 'AGROUND',
  ENGAGED_IN_FISHING = 'ENGAGED_IN_FISHING',
  UNDER_WAY_SAILING = 'UNDER_WAY_SAILING',
  AIS_SART = 'AIS_SART',
  UNKNOWN = 'UNKNOWN',
}

export enum OperationalStatus {
  OPERATIONAL = 'OPERATIONAL',
  DEGRADED = 'DEGRADED',
  DAMAGED = 'DAMAGED',
  DESTROYED = 'DESTROYED',
  INACTIVE = 'INACTIVE',
  UNKNOWN = 'UNKNOWN',
}

export enum DamageAssessment {
  NONE = 'NONE',
  LIGHT = 'LIGHT',
  MODERATE = 'MODERATE',
  HEAVY = 'HEAVY',
  DESTROYED = 'DESTROYED',
  UNKNOWN = 'UNKNOWN',
}

export enum Affiliation {
  FRIENDLY = 'FRIENDLY',
  HOSTILE = 'HOSTILE',
  NEUTRAL = 'NEUTRAL',
  UNKNOWN = 'UNKNOWN',
  ASSUMED_FRIENDLY = 'ASSUMED_FRIENDLY',
  SUSPECT = 'SUSPECT',
  PENDING = 'PENDING',
}

export enum CharacterizationState {
  ASSESSED = 'ASSESSED',
  ASSUMED = 'ASSUMED',
  SUSPECTED = 'SUSPECTED',
  UNCHARACTERIZED = 'UNCHARACTERIZED',
}

/**
 * Ordered classification levels for comparison queries.
 * Lower index = lower classification.
 */
export const CLASSIFICATION_ORDER: readonly Classification[] = [
  Classification.UNCLASSIFIED,
  Classification.CONFIDENTIAL,
  Classification.SECRET,
  Classification.TOP_SECRET,
] as const;

/**
 * Returns the numeric rank of a classification level (0-based).
 */
export function classificationRank(classification: Classification): number {
  const index = CLASSIFICATION_ORDER.indexOf(classification);
  return index === -1 ? 0 : index;
}

File: libs/common-go/models/enums.go
package models

// EntityType mirrors libs/shared-models/src/entity.ts EntityType enum.
const (
	EntityTypeUnknown   = "UNKNOWN"
	EntityTypePerson    = "PERSON"
	EntityTypeVehicle   = "VEHICLE"
	EntityTypeVessel    = "VESSEL"
	EntityTypeAircraft  = "AIRCRAFT"
	EntityTypeFacility  = "FACILITY"
	EntityTypeEquipment = "EQUIPMENT"
	EntityTypeUnit      = "UNIT"
	EntityTypeSignal    = "SIGNAL"
	EntityTypeCyber     = "CYBER"
	EntityTypeSensor    = "SENSOR"
	EntityTypeSatellite = "SATELLITE"
	EntityTypeDrone     = "DRONE"
)

// EntitySource mirrors libs/shared-models/src/entity.ts EntitySource enum.
const (
	EntitySourceHUMINT    = "HUMINT"
	EntitySourceSIGINT    = "SIGINT"
	EntitySourceGEOINT    = "GEOINT"
	EntitySourceOSINT     = "OSINT"
	EntitySourceMASINT    = "MASINT"
	EntitySourceCyber     = "CYBER"
	EntitySourceManual    = "MANUAL"
	EntitySourceAIS       = "AIS"
	EntitySourceADSB      = "ADS_B"
	EntitySourceLINK16    = "LINK16"
	EntitySourceGPS       = "GPS"
	EntitySourceRADAR     = "RADAR"
	EntitySourceCELESTRAK = "CELESTRAK"
	EntitySourceOPENSKY   = "OPENSKY"
	EntitySourceADSBLOL   = "ADSB_LOL"
)

// Classification mirrors libs/shared-models/src/common.ts Classification enum.
const (
	ClassificationUnclassified = "UNCLASSIFIED"
	ClassificationConfidential = "CONFIDENTIAL"
	ClassificationSecret       = "SECRET"
	ClassificationTopSecret    = "TOP_SECRET"
)

// Affiliation mirrors libs/shared-models/src/entity.ts Affiliation enum.
const (
	AffiliationFriendly       = "FRIENDLY"
	AffiliationHostile        = "HOSTILE"
	AffiliationNeutral        = "NEUTRAL"
	AffiliationUnknown        = "UNKNOWN"
	AffiliationAssumedFriendly = "ASSUMED_FRIENDLY"
	AffiliationSuspect        = "SUSPECT"
	AffiliationPending        = "PENDING"
)

// AgeoutState represents the lifecycle state of an entity.
const (
	AgeoutStateLive    = "LIVE"
	AgeoutStateStale   = "STALE"
	AgeoutStateAgedOut = "AGED_OUT"
)

// TrackEnvironment represents the environment in which an entity operates.
const (
	TrackEnvAir        = "AIR"
	TrackEnvSeaSurface = "SEA_SURFACE"
	TrackEnvSubsurface = "SUBSURFACE"
	TrackEnvGround     = "GROUND"
	TrackEnvSpace      = "SPACE"
	TrackEnvUnknown    = "UNKNOWN"
)

// TrackProcessingState represents the processing state of a track.
const (
	TrackProcessingLive         = "LIVE"
	TrackProcessingPredicted    = "PREDICTED"
	TrackProcessingDeadReckoned = "DEAD_RECKONED"
	TrackProcessingHypothesized = "HYPOTHESIZED"
	TrackProcessingHistorical   = "HISTORICAL"
)

// OperationalStatus represents the operational state of an entity.
const (
	OpStatusOperational = "OPERATIONAL"
	OpStatusDegraded    = "DEGRADED"
	OpStatusDamaged     = "DAMAGED"
	OpStatusDestroyed   = "DESTROYED"
	OpStatusInactive    = "INACTIVE"
	OpStatusUnknown     = "UNKNOWN"
)
